src/sentry/integrations/slack/unfurl/discover.py

```diff
@@ -0,0 +1,107 @@
+from sentry.charts.types import ChartType
+from sentry.integrations.slack.utils import logger
+from sentry.models import ApiKey
+
+from . import Handler, UnfurlableUrl, UnfurledUrl
+
+# The display modes on the frontend are defined in app/utils/discover/types.tsx
+display_modes: Mapping[str, ChartType] = {
+    "default": ChartType.SLACK_DISCOVER_TOTAL_PERIOD,
+    "daily": ChartType.SLACK_DISCOVER_TOTAL_DAILY,
+    "top5": ChartType.SLACK_DISCOVER_TOP5_PERIOD,
+    "dailytop5": ChartType.SLACK_DISCOVER_TOP5_DAILY,
+    # TODO(epurkhiser): Previous period
+}
+
+TOP_N = 5
+
+
+def unfurl_discover(data, integration, links: List[UnfurlableUrl]) -> UnfurledUrl:
+    orgs_by_slug = {org.slug: org for org in integration.organizations.all()}
+    unfurls = {}
+
+    for link in links:
+        org_slug = link.args["org_slug"]
+        org = orgs_by_slug.get(org_slug)
+
+        # If the link shared is an org w/o the slack integration do not unfurl
+        if not org:
+            continue
+        if not features.has("organizations:chart-unfurls", org):
+            continue
+
+        params = link.args["query"]
+        params.setlist("order", params.get("sort"))
+        display_mode = str(link.args["query"].get("display", "default"))
+
+        if "daily" in display_mode:
+            params.setlist("interval", ["1d"])
+        if "top5" in display_mode:
+            params.setlist("topEvents", [f"{TOP_N}"])
+
+        resp = client.get(
+            auth=ApiKey(organization=org, scope_list=["org:read"]),
+            path=f"/organizations/{org_slug}/events-stats/",
+            params=params,
+        )
+
+        chart_data = {"seriesName": link.args["query"].get("yAxis", "count()"), "stats": resp.data}
+
+        style = display_modes.get(display_mode, display_modes["default"])
+
+        try:
+            url = generate_chart(style, chart_data)
+        except RuntimeError as exc:
+            logger.error(
+                "Failed to generate chat for discover unfurl: %s",
+                str(exc),
+                exc_info=True,
+            )
+            continue
+
+        unfurls[link.url] = {
+            "blocks": [
+                {
+                    "type": "image",
+                    "title": {
+                        "type": "plain_text",
+                        "text": link.args["query"].get("name", "Dashboards query"),
+                        "emoji": True,
+                    },
+                    "image_url": url,
+                    "alt_text": "Discover Chart",
+                }
+            ]
+        }
+
+    return unfurls
+
+
+def map_discover_query_args(url: str, args: Mapping[str, str]) -> Mapping[str, Any]:
+    """
+    Extracts discover arguments from the discover link's query string
+    """
+    parsed_url = urlparse(url)
+    query = QueryDict(parsed_url.query).copy()
+
+    # Remove some unused query keys
+    query.pop("widths")
+    query.pop("amp")
+
+    return dict(**args, query=query)
```

Comment:
@EvanPurkhiser are you sure `query` can never be part of `args`?

Reply:
Yeah, args comes from the matcher's regex, which is just `org_slug` right now